File: BPMOPMobile_VietNamAirLines/BPMOPMobile_VietNamAirLines/BPMOPMobile.Droid.Core/Component/ControlBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.Graphics;
using Android.OS;
using Android.Runtime;
using Android.Support.V4.Content;
using Android.Support.V4.Content.Res;
using Android.Text;
using Android.Util;
using Android.Views;
using Android.Widget;

namespace BPMOPMobile.Droid.Core.Component
{
    public class ControlBase : ComponentBase
    {
        public LinearLayout _lnContent, _lnLine;
        public TextView _tvTitle, _tvValue;
        public Button _btnAction;
        public Activity _mainAct;

        public ControlBase(Activity _mainAct)
        {
            this._mainAct = _mainAct;
        }

        public override void InitializeCategory(int flagCategory)
        {
            base.InitializeCategory(flagCategory);
        }

        /// <summary>
        /// Cấu hình component mặc định gồm tiều đề và giá trị
        /// </summary>
        public override void InitializeComponent()
        {
            _tvTitle = new TextView(_mainAct);
            _tvValue = new TextView(_mainAct);
            _lnLine = new LinearLayout(_mainAct);
            _lnContent = new LinearLayout(_mainAct);
            _btnAction = new Button(_mainAct);

            _tvTitle.SetTextSize(ComplexUnitType.Sp, 12);
            _tvTitle.SetTextColor(new Color(ContextCompat.GetColor(_mainAct, Resource.Color.clBottomDisable)));
            _tvTitle.SetTypeface(ResourcesCompat.GetFont(_mainAct, Resource.Font.fontarial), TypefaceStyle.Normal);
            _tvTitle.Ellipsize = TextUtils.TruncateAt.End;

            _tvValue.SetTextSize(ComplexUnitType.Sp, 15);
            _tvValue.SetTextColor(new Color(ContextCompat.GetColor(_mainAct, Resource.Color.clBlack)));
            _tvValue.SetTypeface(ResourcesCompat.GetFont(_mainAct, Resource.Font.fontarial), TypefaceStyle.Normal);
            _tvValue.Ellipsize = TextUtils.TruncateAt.End;

            _lnLine.SetBackgroundColor(new Color(ContextCompat.GetColor(_mainAct, Resource.Color.clControlGrayLight)));

            _lnContent.Orientation = Orientation.Vertical;
            _lnLine.Orientation = Orientation.Vertical;
           
        }
        /// <summary>
        /// Vẽ các itewm len LinearLayout
        /// </summary>
        /// <param name="frame"></param>
        public override void InitializeFrameView(LinearLayout frame)
        {
            base.InitializeFrameView(frame);

            frame.RemoveAllViews();
            frame.SetPadding(0, 0, 0, 0);

            if (_lnContent != null && _lnLine != null && _tvTitle != null && _tvValue != null)
            {
                _lnContent.LayoutParameters = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MatchParent, LinearLayout.LayoutParams.WrapContent);
                _lnLine.LayoutParameters = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MatchParent, (int)TypedValue.ApplyDimension(ComplexUnitType.Px, 0, _mainAct.Resources.DisplayMetrics)); // Bỏ Line vì không cần nữa
                _tvTitle.LayoutParameters = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MatchParent, LinearLayout.LayoutParams.WrapContent);
                _tvValue.LayoutParameters = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MatchParent, LinearLayout.LayoutParams.WrapContent);

                int _padding = (int)TypedValue.ApplyDimension(ComplexUnitType.Dip, 3, _mainAct.Resources.DisplayMetrics);

                _lnContent.SetPadding(_padding, 2 * _padding, _padding, 2 * _padding);
                _tvTitle.SetPadding(_padding, 2 * _padding, _padding, _padding);
                _tvValue.SetPadding(_padding, _padding / 2, _padding, 2 * _padding);
                _lnLine.SetPadding(_padding, 2 * _padding, _padding, 2 * _padding);

                _lnContent.AddView(_tvTitle);
                _lnContent.AddView(_tvValue);
                _lnContent.AddView(_lnLine);
                ////_lnContent.AddView(_btnAction); // Android khong cần

                frame.AddView(_lnContent);
            }
        }
    }
}

File: VNABPMOP/app/src/main/java/com/vuthao/bpmop/core/component/ControlBase.java
package com.vuthao.bpmop.core.component;

import android.app.Activity;
import android.graphics.Typeface;
import android.text.TextUtils;
import android.util.TypedValue;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.core.content.ContextCompat;
import androidx.core.content.res.ResourcesCompat;

import com.vuthao.bpmop.R;

public class ControlBase extends ComponentBase {
    protected LinearLayout lnContent, lnLine;
    protected TextView tvTitle, tvValue;
    protected Button btnAction;
    protected Activity mainAct;

    public ControlBase(Activity mainAct) {
        this.mainAct = mainAct;
    }

    @Override
    public void initializeCategory(int Category) {
        super.initializeCategory(Category);
    }

    @Override
    public void initializeComponent() {
        tvTitle = new TextView(mainAct);
        tvValue = new TextView(mainAct);
        lnLine = new LinearLayout(mainAct);
        lnContent = new LinearLayout(mainAct);
        btnAction = new Button(mainAct);

        tvTitle.setTextSize(TypedValue.COMPLEX_UNIT_SP, 12);
        tvTitle.setTextColor(ContextCompat.getColor(mainAct, R.color.clBottomDisable));
        tvTitle.setTypeface(ResourcesCompat.getFont(mainAct, R.font.fontarial), Typeface.NORMAL);
        tvTitle.setEllipsize(TextUtils.TruncateAt.END);

        tvValue.setTextSize(TypedValue.COMPLEX_UNIT_SP, 15);
        tvValue.setTextColor(ContextCompat.getColor(mainAct, R.color.clBlack));
        tvValue.setTypeface(ResourcesCompat.getFont(mainAct, R.font.fontarial), Typeface.NORMAL);
        tvValue.setEllipsize(TextUtils.TruncateAt.END);

        lnLine.setBackgroundColor(ContextCompat.getColor(mainAct,R.color.clControlGrayLight));

        lnContent.setOrientation(LinearLayout.VERTICAL);
        lnLine.setOrientation(LinearLayout.VERTICAL);
    }

    @Override
    public void initializeFrameView(LinearLayout frame) {
        super.initializeFrameView(frame);

        frame.setPadding(0,0,0,0);

        if (lnContent != null && lnLine != null && tvTitle != null && tvValue != null) {
            lnContent.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT));
            lnLine.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, (int)TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_PX, 0 , mainAct.getResources().getDisplayMetrics())));
            tvTitle.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT));
            tvValue.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT));

            int padding = (int)TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_PX, 3 , mainAct.getResources().getDisplayMetrics());

            lnContent.setPadding(padding, 2 * padding, padding, 2 * padding);
            tvTitle.setPadding(padding, 2 * padding, padding, padding);
            tvValue.setPadding(padding, padding / 2, padding, 2 * padding);
            lnLine.setPadding(padding, 2 * padding, padding, 2 * padding);

            lnContent.addView(tvTitle);
            lnContent.addView(tvValue);
            lnContent.addView(lnLine);

            frame.addView(lnContent);
        }
    }

    public interface ControlAttachmentHorizontalListener {
        void OnClick(int pos);
    }

    public interface ControlAttachmentVerticalListener {
        void OnClick(int pos);
    }

    public interface ControlLinkedWorkflowListener {
        void OnDeleteItemClick(int pos);
    }
}
